dev/io.openliberty.http.monitor_fat/fat/src/io/openliberty/http/monitor/fat/RestApplicationTest.java

```diff
@@ -0,0 +1,332 @@
+
+    }
+
+    @Test
+    public void r1_normalPathPost() throws Exception {
+        assertTrue(server.isStarted());
+
+        String route = SIMPLE_RESOURCE_URL + "/pathPost";
+        String requestMethod = HttpMethod.POST;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void r1_normalPathPut() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = SIMPLE_RESOURCE_URL + "/pathPut";
+        String requestMethod = HttpMethod.PUT;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void r1_normalPathDelete() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = SIMPLE_RESOURCE_URL + "/pathDelete";
+        String requestMethod = HttpMethod.DELETE;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void r1_normalPathOptions() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = SIMPLE_RESOURCE_URL + "/pathOptions";
+        String requestMethod = HttpMethod.OPTIONS;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+
+    }
+
+    @Test
+    public void r1_normalPathHead() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = SIMPLE_RESOURCE_URL + "/pathHead";
+        String requestMethod = HttpMethod.HEAD;
+        String responseStatus = "200";
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetric(getVendorMetrics(server), route, responseStatus, requestMethod));
+    }
+
+    @Test
+    @AllowedFFDC
+    public void r1_failDivZero() throws Exception {
+
+        assertTrue(server.isStarted());
+
+        String route = FAIL_RESOURCE_URL + "/zero";
+        String requestMethod = HttpMethod.GET;
+        String responseStatus = "500";
+        String errorType = responseStatus;
+
+        String res = requestHttpServlet(route, server, requestMethod);
+
+        assertTrue(validatePrometheusHTTPMetricWithErrorType(getVendorMetrics(server), route, responseStatus, requestMethod, errorType));
+
+    }
+
+    @Test
+    @AllowedFFDC
```

Comment:
No FFDC is created when 404, right?

Reply:
https://github.com/OpenLiberty/open-liberty/pull/28565/commits/3b17a4d12b48f93656ea09ed58209ff55e7c6843